test/test_bottom_up_iterator.jl

```diff
@@ -0,0 +1,52 @@
+@testset verbose=true "Bottom-up iterator" begin
```

Comment:
could we add a test testing whether the `DepthIterator` returns programs in the right order?